src/ucp/core/ucp_request.h

```diff
@@ -121,8 +121,11 @@ struct ucp_request {
             void                    *buffer;    /* Send buffer */
             ucp_datatype_t          datatype;   /* Send type */
             size_t                  length;     /* Total length, in bytes */
-            ucs_memory_type_t       mem_type;   /* Memory type */
-            ucp_send_nbx_callback_t cb;         /* Completion callback */
+            union {
+                ucp_send_nbx_callback_t cb;         /* Completion callback */
+                ucp_request_t           *super_req; /* Super request that is used
```

Comment:
maybe super_req should be union with user_data? seems more logically close to it

Reply:
done

Reply:
can we remove it from here?

Reply:
> can we remove it from here?

yes, this is leftover